src/cmds/restic/cmd_snapshots.go

```diff
@@ -132,5 +148,49 @@ func runSnapshots(opts SnapshotOptions, gopts GlobalOptions, args []string) erro
 
 	tab.Write(os.Stdout)
 
+	return
+}
+
+//Snapshot provides struct to store snapshots
+type Snapshot struct {
+	ID          string   `json:"id"`
+	Date        string   `json:"date"`
+	Host        string   `json:"host"`
+	Tags        []string `json:"tags"`
+	Directories []string `json:"directories"`
+}
+
+//printSnapshotsJSON provides machine redability
+func printSnapshotsJSON(list []*restic.Snapshot) error {
+
+	enc := json.NewEncoder(os.Stdout)
+	//set JSON prefix
+	fmt.Fprint(os.Stdout, "[")
```

Comment:
Are you aware that you can just pass a `[]Snapshot` to the encoder? Like this:
```go
err := json.NewEncoder(os.Stdout).Encode(list)
```

The encoder prints the list delimiters (`[` and `]`) automatically.